listener.go

```diff
@@ -14,6 +14,14 @@ const (
 	defaultSubscriptionName = "default"
 )
 
+// DeferMessageError is an error type to indicate a message is to be deferred
+type DeferMessageError struct{}
```

Comment:
This is similiar to my old strawman request here
https://github.com/keikumata/azure-pub-sub/pull/13/files

Kind of liked the interface so we could say how long to defer but okay with this too.